samples/arch/smp/pi/sample.smp.pi fails on a number esp32 platforms

`./scripts/twister -a xtensa -s samples/arch/smp/pi/sample.smp.pi -l`

Gets:

```
ERROR   - olimex_esp32_evb          samples/arch/smp/pi/sample.smp.pi                  FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/olimex_esp32_evb/samples/arch/smp/pi/sample.smp.pi/build.log
INFO    - Total complete:    9/  19  47%  skipped:    8, failed:    0
ERROR   - esp_wrover_kit            samples/arch/smp/pi/sample.smp.pi                  FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/esp_wrover_kit/samples/arch/smp/pi/sample.smp.pi/build.log
INFO    - Total complete:   10/  19  52%  skipped:    8, failed:    0
ERROR   - heltec_wifi_lora32_v2     samples/arch/smp/pi/sample.smp.pi                  FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/heltec_wifi_lora32_v2/samples/arch/smp/pi/sample.smp.pi/build.log
INFO    - Total complete:   11/  19  57%  skipped:    8, failed:    0
ERROR   - esp32                     samples/arch/smp/pi/sample.smp.pi                  FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/esp32/samples/arch/smp/pi/sample.smp.pi/build.log
INFO    - Total complete:   12/  19  63%  skipped:    8, failed:    0
ERROR   - odroid_go                 samples/arch/smp/pi/sample.smp.pi                  FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/odroid_go/samples/arch/smp/pi/sample.smp.pi/build.log

```

```
_go/samples/arch/smp/pi/sample.smp.pi/zephyr && /usr/bin/cmake -E echo
/opt/zephyr-sdk-0.14.2/xtensa-espressif_esp32_zephyr-elf/bin/../lib/gcc/xtensa-espressif_esp32_zephyr-elf/10.3.0/../../../../xtensa-espressif_esp32_zephyr-elf/bin/ld.bfd: zephyr/zephyr_pre0.elf section `noinit' will not fit in region `dram0_1_seg'
/opt/zephyr-sdk-0.14.2/xtensa-espressif_esp32_zephyr-elf/bin/../lib/gcc/xtensa-espressif_esp32_zephyr-elf/10.3.0/../../../../xtensa-espressif_esp32_zephyr-elf/bin/ld.bfd: region `dram0_1_seg' overflowed by 17456 bytes
collect2: error: ld returned 1 exit status
```

--- a/scripts/pylib/twister/twisterlib/runner.py
+++ b/scripts/pylib/twister/twisterlib/runner.py
@@ -218,7 +218,7 @@ class CMake:
                     log.write(log_msg)
 
             if log_msg:
-                overflow_found = re.findall("region `(FLASH|ROM|RAM|ICCM|DCCM|SRAM)' overflowed by", log_msg)
+                overflow_found = re.findall("region `(FLASH|ROM|RAM|ICCM|DCCM|SRAM|dram0_1_seg)' overflowed by", log_msg)
                 if overflow_found and not self.options.overflow_as_errors:
                     logger.debug("Test skipped due to {} Overflow".format(overflow_found[0]))
                     self.instance.status = "skipped"

